pkg/nuclide-swift/lib/taskrunner/SwiftPMTaskRunner.js

```diff
@@ -0,0 +1,204 @@
+    const {store, actions} = this._getFlux();
+    return class ExtraUi extends React.Component {
+      props: {
+        activeTaskType: ?string,
+      };
+
+      render(): React.Element<any> {
+        return (
+          <SwiftPMTaskRunnerToolbar
+            store={store}
+            actions={actions}
+            activeTaskType={this.props.activeTaskType}
+          />
+        );
+      }
+    };
+  }
+
+  observeTaskList(callback: (taskList: Array<TaskMetadata>) => mixed): IDisposable {
+    if (this._taskList == null) {
+      this._taskList = Observable.of(SwiftPMTaskRunnerTaskMetadata);
+    }
+    return new DisposableSubscription(
+      this._taskList.subscribe({next: callback}),
+    );
+  }
+
+  getIcon(): ReactClass<any> {
+    return SwiftIcon;
+  }
+
+  runTask(taskName: string): Task {
+    const store = this._getFlux().store;
+    const chdir = store.getChdir();
+    const configuration = store.getConfiguration();
+    const buildPath = store.getBuildPath();
+
+    let command;
+    switch (taskName) {
+      case SwiftPMTaskRunnerBuildTaskMetadata.type:
+        command = buildCommand(
+          chdir,
+          configuration,
+          store.getXcc(),
+          store.getXlinker(),
+          store.getXswiftc(),
+          buildPath,
+        );
+        break;
+      case SwiftPMTaskRunnerTestTaskMetadata.type:
+        command = testCommand(chdir, buildPath);
+        break;
+      default:
+        throw new Error(`Unknown task name: ${taskName}`);
+    }
+
+    atom.commands.dispatch(atom.views.getView(atom.workspace), 'nuclide-console:show');
+    this._logOutput(`${command.command} ${command.args.join(' ')}`, 'log');
+
+    const observable = observeProcess(
+      () => safeSpawn(command.command, command.args),
+    ).do(message => {
+      switch (message.kind) {
+        case 'stderr':
+        case 'stdout':
+          this._logOutput(message.data, 'log');
+          break;
+        case 'exit':
+          if (message.exitCode === 0) {
+            this._logOutput(`${command.command} exited successfully.`, 'success');
+          } else {
+            this._logOutput(`${command.command} failed with exit code ${message.exitCode}`, 'error');
+          }
+          break;
+        default:
+          break;
+      }
+      return message;
+    });
+
+    const task = observableToTaskInfo(observable);
+    invariant(task.observeProgress != null);
+    return {
+      observeProgress: task.observeProgress,
+      onDidComplete: task.onDidComplete,
+      onDidError: task.onDidError,
+      cancel: () => {
+        this._logOutput('Task cancelled.', 'warning');
+        task.cancel();
+      },
+    };
+  }
+
+  getOutputMessages(): Observable<Message> {
+    return this._outputMessages;
+  }
+
+  setProjectRoot(projectRoot: ?Directory): void {
+    const path = projectRoot == null ? '' : projectRoot.getPath();
+    fsPromise.exists(`${path}/Package.swift`).then((fileExists) => {
```

Comment:
nit: I don't think this will lint. (We avoid parens for single arg arrow functions.)

Also, what's the desired behavior for this when your current project isn't a swift one? Not sure what all this would affect, but it seems a little weird for the directory to "stick" to the last selected swift project.